Documentation/openssl.md

```diff
@@ -78,12 +86,42 @@ $ openssl req -new -key apiserver-key.pem -out apiserver.csr -subj "/CN=kube-api
 $ openssl x509 -req -in apiserver.csr -CA ca.pem -CAkey ca-key.pem -CAcreateserial -out apiserver.pem -days 365 -extensions v3_req -extfile openssl.cnf
 ```
 
-## Generate the Kubernetes Worker Keypair
+## Kubernetes Worker Keypairs
+
+This procedure generates a unique TLS certificate for every Kubernetes worker node in your cluster. While unique certificates are less convenient to generate and deploy, they do provide stronger security assurances and the most portable installation experience across multiple cloud-based and on-premises Kubernetes deployments.
+
+### OpenSSL Config
+
+Each worker certificate will need a unique,  minimal openssl config similar to the one used above when creating the api-server certificate. We need to create a configuration file per worker since the `subjectAltName` will be unique per worker. Create one `${WORKER_FQDN}-openssl.cnf` file on your local machine file for every worker in your planned cluster, replacing `${WORKER_FQDN}` with the actual name of each worker.  Now replace the following per-worker values in the appropriate `${WORKER_FQDN}-openssl.cnf` files:
+
+* Replace `${WORKER_IP}`
+* Replace `${WORKER_FQDN}`
+
+**${WORKER_FQDN}-openssl.cnf**
+
+```
+[req]
+req_extensions = v3_req
+distinguished_name = req_distinguished_name
+[req_distinguished_name]
+[ v3_req ]
+basicConstraints = CA:FALSE
+keyUsage = nonRepudiation, digitalSignature, keyEncipherment
+subjectAltName = @alt_names
+[alt_names]
+DNS.1 = ${WORKER_FQDN}
+IP.1 = 127.0.0.1
+IP.2 = ${WORKER_IP}
```

Comment:
couple things:
* If there is a routable hostname, it should technically be the CN rather than a DNS altname. Also, because this is optional, we may want to describe in the section above using a placeholder for WORKER_FQDN (e.g. "kube-worker")
* I feel like we should drop the 127.0.0.1. Generally you will expose insecure api on localhost:8080 anyway.

I also think we can actually make the cnf more generic using env vars. If we change `IP.1 = $ENV::WORKER_IP` we can have a single cnf file and describe command below as:

`WORKER_IP=<replace> openssl req -new -key ${WORKER_FQDN}-worker-key.pem -out ${WORKER_FQDN}-worker.csr -subj "/CN=$WORKER_FQDN" -config worker-openssl.cnf`